react-frontend/src/components/Charts.tsx

```diff
@@ -44,36 +66,67 @@ function Charts({
 		if (limit) verify();
 	}, [limit]);
 
+	useEffect(() => {
+		if (Array.isArray(convergesTo) && convergesTo.length > 0) {
+			setLirecClosedForm(replaceLirecChars());
+		}
+	}, [convergesTo]);
+
 	const wrapExpression = (input: string, label?: string) => {
+		const expr = label ? label.concat(' = ', input) : input;
 		try {
-			const mathy = parse(label ? label.concat(' = ', input) : input).toTex();
+			const mathy = parse(expr).toTex();
 			return `$$${mathy}$$`;
 		} catch (e) {
 			console.log(`failed to parse ${input}`);
+			return expr;
 		}
 	};
 
-	const computeValue = () => {
+	const replaceLirecChars = () => {
 		// we are replacing the exponent operator from python to js syntax
 		// we are also replacing the parentheses with the precision at the end of the expression returned from identify
-		if (convergesTo) {
-			const input = convertConstants(
-				convergesTo
-					.replaceAll('**', '^')
-					.replace(' = 0', '')
-					.replace(/\s\([0-9]+\)$/, '')
-			);
-
-			return wrapExpression(input);
+		let result = new Array<string>();
+		for (const value of convergesTo!!) {
+			const cleanString = value
+				.replaceAll('**', '^')
+				.replace(' = 0', '')
+				.replace(/\s\([0-9]+\)$/, '');
+			const input = convertLirecConstants(cleanString);
+			result.push(wrapExpression(input));
 		}
+		return result;
 	};
-
-	const convertConstants = (input: string) => {
+	// e.g. '(20*alpha_GW - 34)/(alpha_GW + 9)'
+	// should convert to '(20*α[GW] - 34)/(α[GW] + 9)'
+	const convertLirecConstants = (input: string) => {
+		let constantMeta: LirecConstantMetadata[] = [];
 		let cleanString = input;
+		// all of the LIReC constants use just these characters
+		const notConstChars = '[^a-zA-Z0-9_\\[\\[]';
 		for (const c in constants) {
-			if (constants[c].replacement)
-				cleanString = cleanString.replaceAll(c, constants[c].replacement!!);
+			const t1 = new RegExp(`${notConstChars}+${c}${notConstChars}`);
```

Comment:
I don't understand why the 2 regular expressions are required - could this be done with a single regular expression where the last `notConstChars` is optional? Can you add a comment if the 2 expressions are required.

Reply:
Because the only time that it is optional is when it's at the end of the string and I was debugging what the regex needed to be, introduced the second one, and never revisited 😅! Good catch.